tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,554 @@
+
+    _editUserAccount({ store });
+
+    const root = renderUserProfileEdit({ store });
+
+    expect(root.find('.UserProfileEdit-submit-button').dive())
+      .toHaveText('Updating your profile…');
+  });
+
+  it('renders a submit button with a different text when user is not the logged user and editing', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const params = { username: 'another-user' };
+
+    _editUserAccount({ store });
+
+    const root = renderUserProfileEdit({ params, store });
+
+    expect(root.find('.UserProfileEdit-submit-button').dive())
+      .toHaveText(`Updating user's profile…`);
+  });
+
+  it('disables the submit button when username is empty', () => {
+    const root = renderUserProfileEdit();
+
+    root.find(`.UserProfileEdit-username`).simulate(
+      'change',
+      createFakeEventChange({
+        name: 'username',
+        value: '',
+      })
+    );
+
+    expect(root.find('.UserProfileEdit-submit-button'))
+      .toHaveProp('disabled', true);
+  });
+
+  it('dispatches editUserAccount action with new field values on submit', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    const root = renderUserProfileEdit({ errorHandler, store });
+    const user = getCurrentUser(store.getState().users);
+
+    // We want to make sure dispatched action uses the values updated by the user.
+    const location = 'new location';
+    root.find(`.UserProfileEdit-location`).simulate(
+      'change',
+      createFakeEventChange({
+        name: 'location',
+        value: location,
+      })
+    );
+    root.find('.UserProfileEdit-form').simulate('submit', createFakeEvent());
+
+    sinon.assert.calledWith(dispatchSpy, editUserAccount({
+      errorHandlerId: errorHandler.id,
+      userFields: {
+        biography: user.biography,
+        display_name: user.displayName,
+        homepage: user.homepage,
+        location,
+        occupation: user.occupation,
+        username: user.username,
+      },
+      userId: user.id,
+    }));
+  });
+
+  it('renders a success message when user profile has been updated', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const user = getCurrentUser(store.getState().users);
+
+    const occupation = 'new occupation';
+
+    _editUserAccount({
+      store,
+      userFields: {
+        occupation,
+      },
+      userId: user.id,
+    });
+
+    const root = renderUserProfileEdit({ store });
+
+    expect(root.find(Notice)).toHaveLength(0);
+    expect(root.find('.UserProfileEdit-submit-button'))
+      .toHaveProp('disabled', true);
+
+    // The user profile has been updated.
+    store.dispatch(finishEditUserAccount());
+
+    const { isEditing } = store.getState().users;
+    root.setProps({ isEditing });
+
+    expect(root.find(Notice)).toHaveLength(1);
```

Comment:
Is it worth checking the prop of the `Notice` to see that it has the expected message and also perhaps that it has the expected `type`?